data-reconciliation-app/test/e2e-test/src/api.ts

```diff
@@ -48,9 +45,39 @@ export default class Api {
             throw new Error(`🛑 [TEST FAILURE]: Unexpected status code: ${result.status}`);
         }
 
-        console.log(`✅ [PASS] [${result.status} : ${result.statusText}] - ${member.name}\n`);
+        console.log(`✅ [PASS] [${result.status} : ${result.statusText}] - ${member.name}`);
         console.table(result.data.content);
 
         return result.data.content;
     }
+
+    public static async checkPostResponseResult(apiUrl: string, member: DemoMemberProps, expectedStatus: number, testMessage: string) {
```

Comment:
Would it be a better name if we call it and merge the two methods into one?

Also, I think `API` class is mainly describing and running the endpoints. This method is more like a validator.

```typescript
export interface ValidationProps {
   url: string;
   method: 'GET' | 'POST';
   member: DemoMemberProps;
   expectedStatus: number;
   testMessage: string;
}

export class Validator {
   public static async validateRequest(props: ValidationProps) {
      //
      const status = await axios({
         method: props.method,
         url: props.url,
         data: props.member.data,
         httpsAgent: props.member.httpsAgent,
         validateStatus: function (status) {
            return status < 500; // Resolve only if the status code is less than 500
        },
       });

     if (result.status !== expectedStatus) {
         throw new Error(`🛑 [TEST FAILURE]: ${member.name} - ${testMessage}: ${expectedStatus} expected, but got ${result.status}`);
     }

     console.log(`✅ [PASS] ${member.name} - ${testMessage}`);
   }
}
```

Reply:
done